core/src/main/scala/org/apache/spark/scheduler/JobWaiter.scala

```diff
@@ -61,7 +63,10 @@ private[spark] class JobWaiter[T](
     }
   }
 
-  override def jobFailed(exception: Exception): Unit =
-    jobPromise.failure(exception)
+  override def jobFailed(exception: Exception): Unit = {
+    if (!jobPromise.tryFailure(exception)) {
+      logWarning("Ignore failure", exception)
```

Comment:
The first exception will be reported to Promise. Other exceptions will just be logged.